docs/src/callbacks.md

```diff
@@ -0,0 +1,102 @@
+# Callbacks
+Many of the advanced features of Trixi, such as adaptive mesh refinement, are implemented as
+callbacks. A callback is an algorithmic entity that gets passed to the ODE solver and
+is called at specific points during execution to perform certain tasks. Callbacks in Trixi are
+either called after each time step (*step callbacks*) or after each stage of the ODE
+solver (*stage callbacks*).
+
+![callbacks_illustration](https://user-images.githubusercontent.com/65298011/107779585-dc3ba700-6d45-11eb-8087-15afc1cd5080.png)
+
+The advantage of callbacks over hard-coding all features is that it allows to extend Trixi without
+modyfing the internal source code. Trixi provides callbacks for time step
+control, adaptive mesh refinement, I/O and more.
+
+## Step Callbacks
+
+### CFL-based Time Step Control
+Time step control can be performed with a [`StepsizeCallback`](@ref). An example making use
+of this can be found at [examples/2d/elixir\_advection\_basic.jl](https://github.com/trixi-framework/Trixi.jl/blob/main/examples/2d/elixir_advection_basic.jl)
+
+### Adaptive Mesh Refinement
+Trixi uses a hierarchical Cartesian mesh which can be locally refined in a solution-adaptive way.
+This can be used to speed up simulations with minimal loss in overall accuracy. AMR can be used by
```

Comment:
```suggestion
This can be used to speed up simulations with minimal loss in overall accuracy. Adaptive mesh refinement (AMR) can be used by
```

Reply:
Done in fe783972770345b30689982be2f4a293eeffd6c9.